format/spec.md

```diff
@@ -1473,7 +1473,10 @@ The following table describes the possible values for the some of the field with
 
 ### Table Metadata and Snapshots
 
-Table metadata is serialized as a JSON object according to the following table. Snapshots are not serialized separately. Instead, they are stored in the table metadata JSON.
+Table metadata is serialized as a JSON object according to the following table. Snapshots are not serialized separately. Instead, they are stored in the table metadata JSON. 
+
+A metadata JSON file name must end in `.metadata.json`. A metadata JSON file may be compressed with [GZIP](https://datatracker.ietf.org/doc/html/rfc1952). A GZIP compressed file name must end with `.gz.metadata.json`.
```

Comment:
Why should there be any requirements for file naming, except for the FS table scheme that is deprecated? While we need to know what compression is used, it should be part of the implementation details section, not part of the spec itself. This can easily be a responsibility of the catalog or determined by file magic bytes.

Reply:
Lets consolidate discussion on https://github.com/apache/iceberg/pull/12598#discussion_r2070480860

Reply:
Ah, @RussellSpitzer is right that it does imply a requirement for file names. I don't think my intent was to require a specific format -- in fact we violate it to add `gz` -- but it is precedent for specific naming requirements in the spec.

My preference for how to move forward doesn't change though. We don't need a hard requirement here and we can't guarantee one given existing implementations. An implementation note is appropriate for recommendations and to document existing behavior, but we don't need to increase the requirements on writers.

Reply:
Sounds good. updated.